File: 임준형/kotlin-grammer/src/lec08/Lec08Main.java
package lec08;

public class Lec08Main {

    public int max(int a, int b) {
        if (a > b) {
            return a;
        }
        return b;
    }

    public static void repeat(String str, int num, boolean useNewLine) {
        for (int i = 1; i <= num; i++) {
            if (useNewLine) {
                System.out.println(str);
            } else {
                System.out.print(str);
            }
        }
    }

    public void repeat(String str, int num) {
        repeat(str, num, true);
    }

    public void repeat(String str) {
        repeat(str, 3, true);
    }

    public static void printAll(String... strings) {
        for (String string : strings) {
            System.out.println(string);
        }
    }
}

File: 김아린/KotlinStudy/src/main/kotlin/Lec08Main.kt
fun main() {
    repeat("Hello", useNewLine = false)
    printNameAndGender(name = "김아린", gender = "F")

    printAll("a", "b", "c")

    val array = arrayOf("a", "b", "c")
    printAll(*array)
}

fun max(a: Int, b: Int) = if (a > b) a else b

fun repeat(
    str: String,
    num: Int = 3,
    useNewLine: Boolean = true
) {
    for (i in 1..num) {
        if (useNewLine) {
            println(str)
        } else {
            println(str)
        }
    }
}

fun printNameAndGender(name: String, gender: String) {
    println(name)
    println(gender)
}

fun printAll(vararg strings: String) {
    for (str in strings) {
        println(str)
    }
}